### PR TITLE
REF: combine isnaobj+isnaobj_old

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/missing.pyx [hunk 1]
```diff
@@ -205,53 +172,19 @@ cpdef ndarray[uint8_t] isnaobj(ndarray arr):
...
-    n = len(arr)
-    result = np.zeros(n, dtype=np.uint8)
-    for i in range(n):
-        val = arr[i]
-        result[i] = (
-            checknull(val)
-            or util.is_float_object(val) and (val == INF or val == NEGINF)
-        )
-    return result.view(np.bool_)
+    return isnaobj(arr, inf_as_na=True)
```
Comment (mroeschke) [inline pandas/_libs/missing.pyx:182]: Is it possible to just call `isnaobj(arr, inf_as_na=True)` where `isnaobj_old(arr)` is called (same for isnaobj2d)?

Comment (jbrockmendel) [reply pandas/_libs/missing.pyx:182]: it is, im planning to do that in a separate pass

**Review by mroeschke — APPROVED** [review]

File: pandas/_libs/tslibs/nattype.pyx [hunk 2]
```diff
@@ -1237,7 +1248,11 @@ cpdef bint is_null_datetimelike(object val, bint inat_is_null=True):
     elif val is c_NaT:
         return True
     elif util.is_float_object(val) or util.is_complex_object(val):
-        return val != val
+        if val != val:
+            return True
+        if inf_as_na:
```
Comment (jreback) [inline pandas/_libs/tslibs/nattype.pyx:1253]: elif but doesn't matter much

**Review by jreback — APPROVED** [review]
should check perf in any case

**Pull request merged**